### ISSUE TITLE
Spectrum with range mod

### ISSUE THREAD

**Issue by lucknum24** [armel/uv-k5-firmware-custom#285]
When I used the range mod in spectrum function, the arrow is on correct location, but the spectrum length is wrong on scale. 
[图片]
Is that my problem?

**Comment by lucknum24** [comment]
![微信图片_20241026182944](https://github.com/user-attachments/assets/8f30193d-6b62-49a1-b6d0-f93ce2f770f2)

**Comment by armel** [comment]
Hi,

Thank you for this issue.

Where is the problem ?
Can you tell me more?

![380414880-8f30193d-6b62-49a1-b6d0-f93ce2f770f2](https://github.com/user-attachments/assets/b2ad9e1e-18cc-482c-886d-57d0e5e4bdd7)

Armel.

**Comment by lucknum24** [comment]
![1](https://github.com/user-attachments/assets/ba76307a-1652-490f-bcc4-928e76c5f4ec)
![2](https://github.com/user-attachments/assets/c5aaf016-c034-4ac5-9aae-76a1238e91b5)
![3](https://github.com/user-attachments/assets/f4cfdf38-8a28-4548-b152-b15fb66224f2)
@armel   Sorry, I can only speak a little English. Could you understand what I said?
Thank you for your reply.

**Comment by DL2MF** [comment]
Hi Armel,

long time not seen ;-) I can confirm that I noticed also above issue. The marker ist at the correct position, but the spectrum above does not scale to the complete screen width (seems like scale factor calculation / lines per kHz) fails.

**Comment by armel** [comment]
Hello,

Ok,
I see. I don't have a solution for the moment.
I tag this issue as a bug.

Armel.

**Comment by armel** [comment]
@lucknum24 I'd like to ask you a question. How do you get a frequency step of x40 or x160?

**Comment by lucknum24** [comment]
@armel 
![image](https://github.com/user-attachments/assets/2741fb5c-a8b4-4088-acf8-e862cc3efcab)
Use the ScnRng mode. If use the normal mode , there are only 16x,32x,64x,128x bars number choices by pressing 4.  
But in ScnRng mode, the initial bars number is calculated by program and can not be changed by pressing 4. Changing the freq step by pressing 1 and 7, the bars number is also changing.
Did I said clearly? I can take a mp4 to describe it.

**Comment by armel** [comment]
@lucknum24 Crystal clear, thank you 😉

**Comment by armel** [comment]
@lucknum24 By the way, I can now reproduce the problem. That's a good thing. Thank you.

**Comment by lucknum24** [comment]
@armel Thanks for your custom firmware. I think the problem is from the original author. I'v used egzumer edition, and some fork edition from egzumer. All have this problem. I hope you can fix it. Thank you very much!

**Comment by lucknum24** [comment]
I remembered that the original author is fagci, who created the spectrum function.

**Comment by armel** [comment]
@lucknum24 That's right. The spectrum analyzer comes from Fagci. It's a special firmware extension. A sort of application within an application. It has its own management of keyboard shortcuts, backlighting and so on. I've made very few modifications on it.

**Comment by fagci** [comment]
Неправильно вы бутерброд едите, колбасой вниз надо.

@fagci <- вот так упомянуть можно, или сослаться в issue в основном репозитории

**Comment by lucknum24** [comment]
@fagci Ok, I see. I apologize for that. Thank you for your correction.

**Comment by armel** [comment]
@fagci Thank you for your intervention and, above all, thank you for your excellent work 🙏🏼

**Comment by armel** [comment]
Fixed by @fagci in next release v3.7.
